code/src/Managers/BotManager.ts

```diff
@@ -58,72 +58,58 @@ export default class BotManager {
     }
 
     public static async OnInteractionButton(interaction: ButtonInteraction) {
-        const messageInfo: IMessageInfo = await DiscordUtils.ParseInteractionToInfo(interaction);
-        if (interaction.customId == 'onboarding_help') {
-            OnboardingHandler.OnPlacer(messageInfo);
-        } else if (interaction.customId == 'onboarding_diplomacy') {
-            OnboardingHandler.OnStartDiplomacyOnboarding(messageInfo);
-        } else if (interaction.customId == 'onboarding_observe') {
-            OnboardingHandler.OnObserver(messageInfo);
-        } else if (interaction.customId == 'onboarding_development') {
-            OnboardingHandler.OnDevelopment(messageInfo);
-        } else if (interaction.customId == 'diplomacy_invite') {
-            DiplomacyHandler.OnInviteButton(messageInfo);
-        } else if (interaction.customId == 'diplomacy_peek') {
-            DiplomacyHandler.OnPeek(messageInfo);
-        } else if (interaction.customId.startsWith('vote_confirm')) {
-            const id = interaction.customId.split('_')[2];
-            VoteHandler.OnCreateConfirm(messageInfo, id);
-        } else if (interaction.customId.startsWith('vote_destroy')) {
-            const id = interaction.customId.split('_')[2];
-            VoteHandler.OnDestroy(messageInfo, id);
-        } else if (interaction.customId.startsWith('vote_choose')) {
-            const split = interaction.customId.split('_');
-            VoteHandler.OnChoose(messageInfo, [split[2]], split[3]);
-        } else if (interaction.customId.startsWith('application')) {
-            const role = interaction.customId.split('_')[1];
-            ApplicationHandler.OnApplicationStart(messageInfo, role as RoleType);
-        } else if (interaction.customId == 'diplomacy_report') {
-            DiplomacyHandler.OnStartReport(messageInfo);
-        } else if (interaction.customId.startsWith('diplomacy_claim')) {
-            const id = interaction.customId.split('_')[2];
-            DiplomacyHandler.OnClaim(messageInfo, id);
-        } else if (interaction.customId == 'nightswatch') {
-            NightsWatchHandler.OnButton(messageInfo);
-        } else if (interaction.customId.startsWith('coordinate_claim')) {
-            ArtHandler.OnClaimPixel(messageInfo, interaction.customId.split('_')[2]);
+    const messageInfo: IMessageInfo = await DiscordUtils.ParseInteractionToInfo(interaction);
+    const id_parts = interaction.customId.split('_');
+    switch(id_parts[0]) {
+        case 'onboarding': {
+            switch(id_parts[1]) {
```

Comment:
Instead of a nested switch statement, create an InteractionHandler and place the inner switch statement in a method there.

Reply:
An InteractionHandler is a good idea! At that point, is there any reason to distinguish between Button, Modal, and ThreadMenu interactions? I feel like we can leave that to the BotManager, and let the InteractionHandler contain methods like onDiplomacyInteraction and onVoteInteraction to reduce the complexity.